site/utils/apiUtils/index.ts

```diff
@@ -107,3 +115,205 @@ export const getReturnPage = (req: NextApiRequest) => {
         ? queryParam
         : null;
 };
+
+export const delay = (ms: number) => new Promise((res) => setTimeout(res, ms));
+
+export const publishToHootsuite = async (socialMediaPosts: SocialMediaPost[], orgId: string) => {
+    await Promise.all(
+        socialMediaPosts
+            .filter((s) => s.status === SocialMediaPostStatus.pending)
+            .map(async (socialMediaPost) => {
+                const refreshTokens = await getParametersByPath(`/social/${orgId}/hootsuite`);
+
+                if (!refreshTokens || (refreshTokens && refreshTokens.Parameters?.length === 0)) {
+                    await upsertSocialMediaPost(
+                        {
+                            ...socialMediaPost,
+                            status: SocialMediaPostStatus.rejected,
+                        },
+                        orgId,
+                    );
+                    logger.debug("Refresh token is required when creating a social media post");
+                }
+                const refreshToken = refreshTokens.Parameters?.find((rt) =>
+                    rt.Name?.includes(`${socialMediaPost.socialAccount}`),
+                );
+                if (!refreshToken) {
+                    await upsertSocialMediaPost(
+                        {
+                            ...socialMediaPost,
+                            status: SocialMediaPostStatus.rejected,
+                        },
+                        orgId,
+                    );
+                    logger.debug("Refresh token is required when creating a social media post");
+                }
+                const [clientId, clientSecret] = await Promise.all([
+                    getParameter(`/social/hootsuite/client_id`),
+                    getParameter(`/social/hootsuite/client_secret`),
+                ]);
+
+                if (!clientId || !clientSecret) {
+                    await upsertSocialMediaPost(
+                        {
+                            ...socialMediaPost,
+                            status: SocialMediaPostStatus.rejected,
+                        },
+                        orgId,
+                    );
+                    logger.debug("clientId and clientSecret must be defined");
+                }
+
+                const credentials = `${clientId.Parameter?.Value || ""}:${clientSecret.Parameter?.Value || ""}`;
+
+                const authToken = `Basic ${Buffer.from(credentials).toString("base64")}`;
+                const responseToken = await fetch(`${HOOTSUITE_URL}oauth2/token`, {
```

Comment:
Could we create a `data/hootsuite.ts` file which contains the functions to individual api calls. As these calls are invoked in a few places - these functions could be re-used

Reply:
I tried to do that but got a winston logger issue

Reply:
done